assets/js/plugin-check-admin.js

```diff
@@ -0,0 +1,53 @@
+( function ( data ) {
+	const checkItButton = document.getElementById( 'plugin-check__submit' );
+	const pluginsList   = document.getElementById( 'plugin-check__plugins-dropdown' );
+
+	// Return early if the elements cannot be found on the page.
+	if ( ! checkItButton || ! pluginsList ) {
+		console.error( 'Missing form elements on page' );
+		return;
+	}
+
+	checkItButton.addEventListener( 'click', (e) => {
+		e.preventDefault();
+
+		const pluginCheckData = new FormData();
+
+		// Collect the data to pass along for generating a check results.
+		pluginCheckData.append( 'action', 'plugin_check_run_checks' );
+		pluginCheckData.append( 'nonce', data.nonce );
+		pluginCheckData.append( 'plugin', pluginsList.value );
+
+		fetch(
+			data.ajaxUrl,
+			{
+				method: 'POST',
+				credentials: 'same-origin',
+				body: pluginCheckData
+			}
+		)
+		.then(
+			( response ) => {
+				if ( ! response.ok ) {
+					throw new Error( `${response.status}: ${response.statusText}` );
+				}
+
+				return response.json();
+			}
+		)
+		.then(
+			( data ) => {
+				if ( ! data.success && ( ! data.data || ! data.data.message ) ) {
+					throw new Error( 'Response contains no data' );
+				}
+
+				console.log( data.data.message );
```

Comment:
Doesn't this need to use `data.data[ 0 ].message`? You had that here before which I thought was right, but now you got rid of the `[ 0 ]`.

I tested this and it is indeed not working. You need to also still include a condition that covers the error case. If the 403 error from the AJAX action is returned, the `data.data[ 0 ].message` in the error response should be logged via `console.error`. That is not the case right now.

Reply:
Thanks @felixarntz I've updated the code based on your feedback with comments to highlight the different scenarios/errors that are captured here.